File: WeatherApp/ios/WeatherApp/NativeHelper.m
//
//  NativeHelper.m
//  WeatherApp
//
//  Created by Prashant Tiwari on 30/04/2023.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(NativeHelper, NSObject)
  RCT_EXTERN_METHOD(isLocationAuthorized:
    (RCTResponseSenderBlock) callback
  )
  + (BOOL)requiresMainQueueSetup
  {
    return NO;
  }
@end

File: WeatherApp/ios/WeatherApp/NativeHelper.swift
//
//  NativeHelper.swift
//  WeatherApp
//
//  Created by Prashant Tiwari on 30/04/2023.
//

import Foundation
import CoreLocation

@objc(NativeHelper) class NativeHelper: NSObject {
  @objc public func isLocationAuthorized(_ callback: RCTResponseSenderBlock) {
    let status = CLLocationManager.authorizationStatus()
    if status == .authorizedAlways || status == .authorizedWhenInUse {
      callback([true])
    } else {
      callback([false])
    }
  }
}

